src/components/ICA/__tests__/ICA.spec.js

```diff
@@ -3,76 +3,75 @@
-
 describe('ICA', () => {
-  let ica;
-  const largeValue = value * total;
-
-  beforeEach(() => {
-    ica = shallow(<ICA {...defaultProps} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(<ICA label="test ICA" total={10} value={5} />, {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    });
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('ICA');
   });
 
-  it('renders', () => {
-    expect(ica).toMatchSnapshot();
+  test('should render en dash when `value` is `null`', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={null} />);
+    expect(queryByText('–')).toBeVisible();
   });
 
-  it('renders the null value', () => {
-    ica.setProps({ value: null });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a large value', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={1000000} />);
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('renders a large value', () => {
-    ica.setProps({ value: largeValue });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a subset of values when `value` and `total` are provided', () => {
+    const { getByText } = render(<ICA label="test ICA" value={5} total={10} />);
+    expect(getByText('5').nextSibling).toBe(getByText(/10/i));
   });
 
-  it('renders a subset of values', () => {
-    ica.setProps({ total });
-
-    expect(ica).toMatchSnapshot();
+  test('should not render subset of values when `total` is close to `value`', () => {
+    const { queryByText } = render(
+      <ICA label="test ICA" value={1000000} total={999995} />
+    );
+    expect(queryByText('999995')).not.toBeInTheDocument();
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('does not render a subset when the total is close to the provided value', () => {
-    ica.setProps({ total: largeValue - value, value: largeValue });
-    expect(ica).toMatchSnapshot();
+  test('should not render `total` if it is the same as `value`', () => {
+    const { getByText } = render(
+      <ICA label="test ICA" value={10} total={10} />
+    );
+    // Expect 2 ICA nodes: the label and the value.
+    expect(getByText(/test ICA/i).parentNode.children.length).toBe(2);
   });
 
-  it('renders with a value that is same as the total value', () => {
-    ica.setProps({ total: value });
-
-    expect(ica).toMatchSnapshot();
-  });
-
-  it('renders a percentage', () => {
-    ica.setProps({ percentage: true, total });
-
-    expect(ica).toMatchSnapshot();
+  test('should render `total` when forced', () => {
+    const { getByText } = render(
+      <ICA label="test ICA" value={10} total={10} forceShowTotal />
+    );
+    // Expect 3 ICA nodes: the label, value, and the total.
+    expect(getByText(/test ICA/i).parentNode.children.length).toBe(3);
```

Comment:
Is it sufficient to use solely `queryByText(' / 10')` here?

Reply:
This would return `null` & not pass the test. There's weirdness with the new line so maybe a matching function could solve it. Just seems easier to check for the correct nodes.

Reply:
Probably better in this case to run a `queryByAll` type of query and check for the number of results: https://testing-library.com/docs/dom-testing-library/api-queries#queryallby